### PR TITLE
add tooltips to disabled buttons

### PR DESCRIPTION
**Pull request by panentheos**
This adds tooltips to most disabled buttons explaining why you can't currently perform the action. This should improve form usability.

### CODE REVIEW

File: src/components/NewWorkspaceModal.js [hunk 1]
```diff
@@ -98,14 +100,18 @@ export default class NewWorkspaceModal extends Component {
...
+      prettify: v => ({ namespace: 'Billing project', name: 'Name' }[v] || validate.prettify(v))
+    })
     return h(Modal, {
       title: cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
       onDismiss,
-      okButton: buttonPrimary({
-        disabled: errors,
-        onClick: () => this.create()
-      }, cloneWorkspace ? 'Clone Workspace' : 'Create Workspace')
+      okButton: h(TooltipTrigger, { content: Utils.summarizeErrors(errors) }, [
```
Comment (zarsky-broad) [inline src/components/NewWorkspaceModal.js:109]: I had to look it up, I feel like we should mention somewhere like in the `TooltipTrigger` docstring that when `content` is falsy we don't show an empty tooltip.

File: src/components/notebook-utils.js [hunk 2]
```diff
@@ -100,32 +102,34 @@ export class NotebookCreator extends Component {
...
+                      this.setState({ creating: false })
+                      reportError('Error creating notebook', error)
+                    }
+                  )
+                }
+              }, 'Create Notebook')
+            ])
           }, [
-            div({ style: Style.elements.sectionHeader }, 'Name'),
+            div({ style: Style.elements.sectionHeader }, 'Name *'),
```
Comment (zarsky-broad) [inline src/components/notebook-utils.js:124]: I may be misremembering, but I believe we decided to mark only optional inputs app-wide?

File: src/pages/StyleGuide.js [hunk 3]
```diff
@@ -65,15 +64,8 @@ class StyleGuide extends Component {
...
-          name: 'input',
-          inputProps: {
-            placeholder: 'ValidatedInput wants an email',
-            value: this.state.validatedInputValue,
-            onChange: e => this.setState({ validatedInputValue: e.target.value, validatedInputTouched: true })
-          },
-          errors: this.state.validatedInputTouched ?
-            validate.single(this.state.validatedInputValue, { email: true }) :
-            null
+          inputProps: { value: 'Hello there!', readOnly: true },
```
Comment (zarsky-broad) [inline src/pages/StyleGuide.js]: Can we make this writable? One of the things that is useful here is to be able to see the validation behavior.

**Review by zarsky-broad — APPROVED** [review]

File: src/components/notebook-utils.js (hunk 2 above)
Comment (panentheos) [reply src/components/notebook-utils.js:124]: There are definitely recent designs showing asterisks for required. Let's follow up with design.

File: src/components/NewWorkspaceModal.js (hunk 1 above)
Comment (panentheos) [reply src/components/NewWorkspaceModal.js:109]: Done.

**Pull request merged**